lib/puppet/defaults.rb

```diff
@@ -823,6 +823,8 @@ def self.initialize_default_settings!(settings)
           it should match `/\A[a-z0-9._-]+\Z/`.)
         * The special value `ca` is reserved, and can't be used as the certname
           for a normal node.
+          
+          **Note:** For full functionality of the `puppet config print` command and other dependent commands, include the certname in the main section of the puppet.conf file.
```

Comment:
It almost sounds like `puppet config print` is the most important thing, but it's more that all of puppet assumes certname is set in the main section. Maybe something like "certname" must be set in the "main" section. Setting it in a different section will cause errors.